lib/global-variable-adapter.js

```diff
@@ -0,0 +1,27 @@
+var createLocalStorageAdapter = require('./local-storage-adapter')
+
+module.exports = createGlobalVariableAdapter
+
+function createGlobalVariableAdapter (queueName) {
+  window.__queueThat__ = window.__queueThat__ || {}
+
+  var localStorageAdapter = createLocalStorageAdapter(queueName)
```

Comment:
Yeh, I think the code could be a lot neater. I am thinking that this idea of a pluggable adapter might be a bit contrived and maybe there should just be one storage module that will fall back to global storage automatically. I will probably rewrite in the future.